src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -179,6 +179,9 @@ public void connect() {
         broken = true;
         throw new JedisConnectionException("Failed connecting to "
             + socketFactory.getDescription(), ex);
+      } catch(Exception ex) {
```

Comment:
What other exceptions can this throw? If we are just generalizing, should we just roll this in the above?

Reply:
> What other exceptions can this throw?

`JedisConnectionException`.

Actually, using `DefaultJedisSocketFactory`, it is never `IOException`. Because `DefaultJedisSocketFactory.createSocket()` wraps the IOException in a JedisConnectionException.

> should we just roll this in the above?

I tried. But the compiler is forcing to handle `IOException` separately. So if we want to roll this in above, `throws IOException` had to be added with the method which I wanted to avoid.

Reply:
> Actually, using DefaultJedisSocketFactory, it is never IOException. Because DefaultJedisSocketFactory.createSocket() wraps the IOException in a JedisConnectionException.

It can throw IOException, the `Socket.connect` can cause that and then we have a catch that just rethrows after attempting to close the socket if it was opened.

> I tried. But the compiler is forcing to handle IOException separately. So if we want to roll this in above, throws IOException had to be added with the method which I wanted to avoid.

I think it should work fine, this works and compile for me:

```java
public void connect() {
  if (!isConnected()) {
    try {
      socket = jedisSocketFactory.createSocket();

      outputStream = new RedisOutputStream(socket.getOutputStream());
      inputStream = new RedisInputStream(socket.getInputStream());
    } catch (Exception ex) {
      broken = true;
      throw new JedisConnectionException("Failed connecting to "
          + jedisSocketFactory.getDescription(), ex);
    }
  }
}
```

You can also replace `Exception` with `IOException | JedisConnectionException` if you want to be explicit about the exact types of exceptions.

Reply:
1. Sorry, I misunderstood.
2. You're right. I probably had some other code that was affecting this code.